x/budget/keeper/budget.go

```diff
@@ -74,7 +74,11 @@ func (k Keeper) CollectibleBudgets(ctx sdk.Context) (budgets []types.Budget) {
 	if params.EpochBlocks > 0 && ctx.BlockHeight()%int64(params.EpochBlocks) == 0 {
 		for _, budget := range params.Budgets {
 			err := budget.Validate()
+			// ^ We can avoid having to validate all budgets at every BeginBlock since
+			// we already know that the budgets are valid from the validation of params
 			if err == nil && !budget.Expired(ctx.BlockTime()) {
+				// ^ We can consider automatically deleting expired budgets, instead of
+				// expecting this to be done through governance proposals.
```

Comment:
Thanks for the suggestion, I think the number of budgets will be relatively small, and even if it is over, the past details may be important, so I think it would be better to leave as ended proposals remain in the gov. and it may be burdensome to change params data without gov IMO.